static/js/message_edit.js

```diff
@@ -854,4 +854,23 @@ exports.handle_narrow_deactivated = function () {
     }
 };
 
+exports.move_message_to_stream = function (message_id, stream_id, topic_name) {
```

Comment:
Let's rename this to `exports.move_topic_containing_message_to_stream`, for clarity about what it does.

And maybe add a comment right here making clear the `stream_id` and `topic_name` are the NEW values, and the old ones are wherever the message is located, since that may not be intuitive.